LibTrinityCore-1.0/LibTrinityCore-1.0.lua

```diff
@@ -39,15 +40,17 @@ end
 
 local frame = CreateFrame("Frame")
 frame:SetScript("OnEvent", function(self, event, prefix, message, channel, sender)
+  PlayerName = table.concat({UnitFullName("player")}, '-') -- "PlayerName-RealmName"
   if event == "PLAYER_ENTERING_WORLD" then
     self:UnregisterEvent("PLAYER_ENTERING_WORLD")
     self:RegisterEvent("CHAT_MSG_ADDON")
-    SendAddonMessage("TrinityCore", "p0000", "WHISPER", (UnitName("player")))
+    C_ChatInfo.RegisterAddonMessagePrefix("TrinityCore")
+    C_ChatInfo.SendAddonMessage("TrinityCore", "p0000", "WHISPER", PlayerName)
     return
   end
   assert(event == "CHAT_MSG_ADDON")
   if prefix ~= "TrinityCore" then return end
-  if sender ~= UnitName("player") then return end
+  if sender ~= PlayerName then return end
```

Comment:
Does this always work? I dont know when client adds -Realm to name and when it doesn't

Reply:
I used `UnitFullName` which always returns the realm name, so yes.

However, both `UnitName` and `UnitFullName` might return the wrong name if the player is transformed by some spell, so instead we should use `UnitNameUnmodified`. Unfortunately, it does not guarantee a realm name like `UnitFullName` does, so I have also added `GetRealmName` to be 100% safe.

Reply:
@Shauren's question was about whether `sender` is guaranteed to be the fullname form here.

Reply:
That seems to be the case in my experience, but I haven't been able to find any 'official' confirmation.

As of the latest commit, I could have it check against both `playerFullName` and `playerName`, though. Should I?